python/ray/includes/gcs_client.pxi

```diff
@@ -334,14 +334,38 @@ cdef class NewGcsClient:
         return raise_or_return(convert_get_all_node_info(status, move(reply)))
 
     def async_get_all_node_info(
-        self, timeout: Optional[float] = None
+        self,
+        filter_node_id: NodeID | None,
+         filter_state: Literal["ALIVE"] | Literal["DEAD"] | None,
```

Comment:
`|` is only supported python 3.10+, use Union instead

Reply:
done. it's also caught by ci